### ISSUE TITLE
Filtering repeats

### ISSUE THREAD

**Issue by Tierhon** [TravisWheelerLab/ULTRA#4]
Hi,
in the output JSON file, you have a score and Log2 Pval for each of the found repeats. Do you have any rules of thumb for which repeats you would keep for further analysis?

Thank you.

Sincerely,
Ole

**Comment by DanielOlson** [comment]
Hi Ole,
Right now the Pval is calculated directly from score, although that might change slightly in the future (and so for the time being I recommend using score cutoffs instead of Pval cutoffs).

I have found in my own work that a score threshold of 8 works fairly well - this will catch things like:
ACACACAGACACACACACA
AAAAAAAAAAAGAGAGAGAG
AAAAAAAAAAAAAAAAAAAAAAAAA
TTTAAAAAAAATTTTTGTTTTTTTTT

and will filter things like:
AAACAAACAAACAAA (too short)
CCTCCTCCCACCCCCCC (poor consensus / too short)
AAAAAAAAAAAA (too short)
ATAACATTATTATACCATACTTTTACTATACCTTTTCTATA (poor consensus)

If you are interested in exclusively longer repeats or repeats that are very definitely repeats and not some sort of protorepeat then I have found a score threshold of 15-20 works very well. This will catch things like:
TATTTTATTTTATTTTATTTTATTTT
TTTATTTTATTTTATTTTATTTTTATTTATTTTTTTT
AAACAAACAAACAAACAAACAAACAAAC

The best method to determine score threshold is to run ULTRA with a very low score threshold on your sequence and then try to gauge what repeats you are interested in and what their minimum score is.

Daniel
